File: elixir-elasticemail-examples/examples/statistics.exs
# Account-wide sending statistics for the last 30 days.
# Dates are ISO 8601 without timezone (YYYY-MM-DDThh:mm:ss), interpreted as UTC.
iso = fn dt -> dt |> DateTime.truncate(:second) |> DateTime.to_iso8601() |> String.slice(0, 19) end

to = DateTime.utc_now()
from = DateTime.add(to, -30 * 24 * 60 * 60, :second)

case ElasticEmail.get("/statistics", from: iso.(from), to: iso.(to)) do
  {:ok, s} ->
    IO.puts("=== Statistics #{iso.(from)} to #{iso.(to)} ===")
    IO.puts("Recipients:    #{s["Recipients"]}")
    IO.puts("Emails total:  #{s["EmailTotal"]}")
    IO.puts("Delivered:     #{s["Delivered"]}")
    IO.puts("Bounced:       #{s["Bounced"]}")
    IO.puts("In progress:   #{s["InProgress"]}")
    IO.puts("Opened:        #{s["Opened"]}")
    IO.puts("Clicked:       #{s["Clicked"]}")
    IO.puts("Unsubscribed:  #{s["Unsubscribed"]}")
    IO.puts("Complaints:    #{s["Complaints"]}")

  error ->
    IO.puts(:stderr, "Error fetching statistics: #{ElasticEmail.format_error(error)}")
    System.halt(1)
end

File: bun-elasticemail-examples/javascript/examples/statistics.js
import { Configuration, StatisticsApi } from "@elasticemail/elasticemail-client-ts-axios";

const config = new Configuration({ apiKey: process.env.ELASTICEMAIL_API_KEY });
const statisticsApi = new StatisticsApi(config);

// Account-wide sending statistics for the last 30 days.
// Dates are ISO 8601 without timezone (YYYY-MM-DDThh:mm:ss), interpreted as UTC.
const iso = (d) => d.toISOString().slice(0, 19);
const to = new Date();
const from = new Date(to.getTime() - 30 * 24 * 60 * 60 * 1000);

try {
  const { data } = await statisticsApi.statisticsGet(iso(from), iso(to));

  console.log(`=== Statistics ${iso(from)} to ${iso(to)} ===`);
  console.log("Recipients:   ", data.Recipients);
  console.log("Emails total: ", data.EmailTotal);
  console.log("Delivered:    ", data.Delivered);
  console.log("Bounced:      ", data.Bounced);
  console.log("In progress:  ", data.InProgress);
  console.log("Opened:       ", data.Opened);
  console.log("Clicked:      ", data.Clicked);
  console.log("Unsubscribed: ", data.Unsubscribed);
  console.log("Complaints:   ", data.Complaints);
} catch (err) {
  console.error("Error fetching statistics:", err.response?.status, err.response?.data ?? err.message);
  process.exit(1);
}
